.github/workflows/cicd.yaml

```
@@ -74,82 +52,161 @@ jobs:
-
-      # ------------------------------------------------------------
-      # Create / update the SVG badge (only on direct pushes to our
-      # main branches, never on PRs/forks so we avoid permission woes)
-      # ------------------------------------------------------------
-      - name: Generate coverage badge
-        if: github.event_name == 'push' && (github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/main')
-        uses: tj-actions/coverage-badge-py@v2
-        with:
-          output: coverage.svg
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.9)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)